src/services/annotations-parser.js

```diff
@@ -17,23 +17,85 @@ export function parseAnnotationSets(manifest, canvasIndex) {
     const annotations = canvas.annotations;
     const duration = Number(canvas.duration);
 
-    if (annotations?.length > 0 && annotations[0].type === 'AnnotationPage') {
-      annotations.map((annotation) => {
-        if (annotation.type === 'AnnotationPage') {
-          let annotationSet = { label: getLabelValue(annotation.label) };
-          if (annotation.items?.length > 0) {
-            annotationSet.items = parseAnnotationItems(annotation.items, duration);
-          } else {
-            annotationSet.url = annotation.id;
-          }
-          annotationSets.push(annotationSet);
-        }
+    annotationSets = parseAnnotationPage(annotations, duration);
+    return { canvasIndex, annotationSets };
+  } else {
+    return null;
+  }
+};
+
+/**
+ * Fetch and parse linked AnnotationPage json file
+ * @param {String} url URL of the linked AnnotationPage .json
+ * @param {Number} duration Canvas duration
+ * @returns {Object} JSON object for the annotations
+ * 
+ */
+export async function parseExternalAnnotationPage(url, duration) {
+  const urlRegex = /^(https?:\/\/)?([\w-]+\.)+[\w-]+(\/[\w\-._~:\/?#[\]@!$&'()*+,;=]*)?\.json$/;
+
+  // Validate given URL
+  if (url == undefined || url.match(urlRegex) == null) {
+    return [];
+  } else {
+    let fileData = null;
+
+    // get file type
+    await fetch(url)
+      .then(handleFetchErrors)
+      .then(function (response) {
+        fileData = response;
+      })
+      .catch((error) => {
+        console.error(
+          'annotations-parser -> parseExternalAnnotationPage() -> fetching transcript -> ',
+          error
+        );
+        return [];
       });
+
+    if (fileData == null) {
+      return [];
+    } else {
+      try {
+        const annotationPage = await fileData.json();
+        const annotations = parseAnnotationPage([annotationPage], duration);
+        return annotations;
+      } catch (e) {
+        console.error(
+          'annotations-parser -> parseExternalAnnotationPage() -> Error: parsing AnnotationPage at, ',
+          url
+        );
+        return [];
+      }
     }
   }
+}
 
-  return { canvasIndex, annotationSets };
-};
+/**
+ * Parse a annotations in a given list of AnnotationPage objects.
+ * @param {Array} annotationPages AnnotationPage from either Canvas or linked .json
+ * @param {Number} duration Canvas duration
+ * @returns {Array<Object>} a parsed list of annotations in the AnnotationPage
+ * [{ label: String, items: Array<Object> }]
+ */
+function parseAnnotationPage(annotationPages, duration) {
```

Comment:
Since this method is handling multiple annotation pages, how about renaming it?
```suggestion
function parseAnnotationPages(annotationPages, duration) {
```
Should the parsing of a single annotation page be split out so `parseExternalAnnotationPage` can all it?